common-service/src/main/java/org/sunbird/controllers/BaseController.java

```diff
@@ -270,4 +276,160 @@ public static void setActorRef(Object actorRefObj) {
+        ServiceBaseGlobal.requestInfo.get(ctx().flash().get(JsonKey.REQUEST_ID));
+    org.sunbird.common.request.Request req = new org.sunbird.common.request.Request();
+
+    Map<String, Object> params = (Map<String, Object>) requestInfo.get(JsonKey.ADDITIONAL_INFO);
+
+    params.put(JsonKey.LOG_TYPE, JsonKey.API_ACCESS);
+    params.put(JsonKey.MESSAGE, "");
+    params.put(JsonKey.METHOD, request.method());
+    // calculate  the total time consume
+    long startTime = (Long) params.get(JsonKey.START_TIME);
+    params.put(JsonKey.DURATION, calculateApiTimeTaken(startTime));
+    removeFields(params, JsonKey.START_TIME);
+    params.put(
+        JsonKey.STATUS, String.valueOf(((Response) response).getResponseCode().getResponseCode()));
+    params.put(JsonKey.LOG_LEVEL, JsonKey.INFO);
+    req.setRequest(
+        generateTelemetryRequestForController(
+            TelemetryEvents.LOG.getName(),
+            params,
+            (Map<String, Object>) requestInfo.get(JsonKey.CONTEXT)));
+    // if any request is coming form /v1/telemetry/save then don't generate the telemetry log
+    // for it.
+    lmaxWriter.submitMessage(req);
+
+    Response courseResponse = (Response) response;
+    if (!StringUtils.isBlank(key)) {
+      Object value = courseResponse.getResult().get(JsonKey.RESPONSE);
+      courseResponse.getResult().remove(JsonKey.RESPONSE);
+      courseResponse.getResult().put(key, value);
+    }
+
+    // remove request info from map
+    ServiceBaseGlobal.requestInfo.remove(ctx().flash().get(JsonKey.REQUEST_ID));
+    return Results.ok(
+        Json.toJson(BaseController.createSuccessResponse(request, (Response) courseResponse)));
+    // }
+    /*
+     * else {
+     *
+     * ProjectCommonException exception = (ProjectCommonException) response; return
+     * Results.status(exception.getResponseCode(),
+     * Json.toJson(BaseController.createResponseOnException(request, exception))); }
+     */
+  }
+
+  private long calculateApiTimeTaken(Long startTime) {
+
+    Long timeConsumed = null;
+    if (null != startTime) {
+      timeConsumed = System.currentTimeMillis() - startTime;
+    }
+    return timeConsumed;
+  }
+
+  private void removeFields(Map<String, Object> params, String... properties) {
+    for (String property : properties) {
+      params.remove(property);
+    }
+  }
+
+  private Map<String, Object> generateTelemetryRequestForController(
+      String eventType, Map<String, Object> params, Map<String, Object> context) {
+
+    Map<String, Object> map = new HashMap<>();
+    map.put(JsonKey.TELEMETRY_EVENT_TYPE, eventType);
+    map.put(JsonKey.CONTEXT, context);
+    map.put(JsonKey.PARAMS, params);
+    return map;
+  }
+
+  /**
+   * This method will create data for success response.
+   *
+   * @param request play.mvc.Http.Request
+   * @param response Response
+   * @return Response
+   */
+  public static Response createSuccessResponse(Request request, Response response) {
```

Comment:
This method does exactly same thing being done by base class createSuccessResponse method.